src/main/java/tech/picnic/jolo/Relation.java

```diff
@@ -26,7 +26,7 @@
  * @param <L> The Java class that the left-hand side of the relation is mapped to
  * @param <R> The Java class that the right-hand side of the relation is mapped to.
  */
-final class Relation<L, R> {
+public final class Relation<L, R> {
```

Comment:
It's because `Pair` is static nested class. We can move it to a separate file. IDK which one is more desirable.

Reply:
If this file does not otherwise need to be public (didn't check) then yes I think moving it out makes sense. The smaller the API surface, the better.